quickstart/dotnet-core/Program.cs

```diff
@@ -207,102 +208,97 @@ public static async Task ForceReconnectAsync()
-            int disposedRetry = 0;
 
+            IDatabase cache = Connection.GetDatabase();
             while (true)
             {
                 try
                 {
-                    return func();
+                    return await func(cache);
                 }
                 catch (Exception ex) when (ex is RedisConnectionException || ex is SocketException)
                 {
                     reconnectRetry++;
                     if (reconnectRetry > RetryMaxAttempts)
                         throw;
-                    await ForceReconnectAsync();
-                }
-                catch (ObjectDisposedException)
-                {
-                    disposedRetry++;
-                    if (disposedRetry > RetryMaxAttempts)
-                        throw;
+                    try
+                    {
+                        await ForceReconnectAsync();
+                        cache = Connection.GetDatabase();
+                    }
+                    catch (ObjectDisposedException) { }
                 }
             }
         }
 
-        public static Task<IDatabase> GetDatabaseAsync()
+        static async Task Main(string[] args)
         {
-            return BasicRetryAsync(() => Connection.GetDatabase());
-        }
+            await InitializeAsync();
 
-        public static Task<System.Net.EndPoint[]> GetEndPointsAsync()
-        {
-            return BasicRetryAsync(() => Connection.GetEndPoints());
-        }
+            // Perform cache operations using the cache object...
+            Console.WriteLine("Running... Press 'q' to quit.");
+            ConsoleKey inputKey = ConsoleKey.A; // Init value, it may be anything different to Q
 
-        public static Task<IServer> GetServerAsync(string host, int port)
-        {
-            return BasicRetryAsync(() => Connection.GetServer(host, port));
-        }
+            while (inputKey != ConsoleKey.Q)
+            {
+                for (int i = 0; i < 2; i++)
+                {
+                    Task.Run(() => RunRedisCommands());
+                    Thread.Sleep(500);
+                }
 
-        static async Task Main(string[] args)
-        {
-            await InitializeAsync();
+                Thread.Sleep(2000);
 
-            IDatabase cache = await GetDatabaseAsync();
+                if (Console.KeyAvailable)
+                {
+                    inputKey = Console.ReadKey().Key;
+                }
+            }
 
-            // Perform cache operations using the cache object...
+            await CloseConnectionAsync(_connection);
+        }
+
+        private static async Task RunRedisCommands()
+        {
 
             // Simple PING command
             string cacheCommand = "PING";
-            Console.WriteLine("\nCache command  : " + cacheCommand);
-            Console.WriteLine("Cache response : " + cache.Execute(cacheCommand).ToString());
+            Console.WriteLine($"\nCache command: " + cacheCommand);
```

Comment:
>$"\nCache command: " + cacheCommand [](http://example.com/codeflow?start=30&length=35)

use string interpolation consistently, rather than +

`$"\nCache command: {cacheCommand}"`